src/redux/actions/syncBattle.ts

```diff
@@ -615,6 +619,14 @@ export const syncBattle = createAsyncThunk<CalcdexBattleState, SyncBattlePayload
                 };
               }
 
+              // Update the battle form if the preset has a different forme
+              // Specifically for Urshifu... this mon is annoying.
+              if (matchedPreset.speciesForme && matchedPreset.speciesForme !== syncedPokemon.speciesForme) {
+                syncedPokemon.speciesForme = matchedPreset.speciesForme;
+                const dexSpecies = dex.species.get(matchedPreset.speciesForme);
+                syncedPokemon.types = [...dexSpecies.types];
+              }
```

Comment:
ah, I see what you were doin here (& also in `getPresetFormes.ts`), which actually helped me find a bug in regards to handling OTS, specifically updating the `speciesForme`. I'll address that separately tho, so no worries. thanks!